kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt

```diff
@@ -1,18 +1,277 @@
+//              brush = Brush.linearGradient(colors = listOf(barSeries.areaColor!!))
+        )
+    }
+
+}
+
+/**
+ * Get position of data point to be plotted in chart.
+ *
+ * @param index Data index.
+ * @param data Chart data.
+ * @param canvasHeight Canvas' height.
+ *
+ * @return Data point's position in canvas.
+ */
+private fun <T> getDataPoint(
+        index: Int,
+        data: ChartData<T>,
+        canvasHeight: Float
+) = Offset(
+        (index + 1) * mHGap,
+        canvasHeight - ((data.value / mMaxVal) * canvasHeight)
+)
+
+/**
+ * Canvas' origin point.
+ *
+ * @return Offset point.
+ */
+private fun DrawScope.origin() = Offset(0f, size.height)
+
+/**
+ * Draw line chart grids.
+ *
+ * @param count Grid count.
+ * @param color Grid color.
+ */
+private fun DrawScope.drawGrids(count: Int, color: Color) {
+    val startPoint = origin()
+//    val pathEffect = PathEffect.dashPathEffect(intervals = floatArrayOf(10f, 10f))
+
+    for (i in 1..count) {
+        // Horizontal lines
+        drawLine(
+                color = color,
+                start = Offset(0f, startPoint.y - i * mVGap),
+                end = Offset(size.width, startPoint.y - i * mVGap),
+//            pathEffect = pathEffect
+        )
+        // Vertical lines
+        drawLine(
+                color = color,
+                start = Offset(i * mHGap, 0f),
+                end = Offset(i * mHGap, size.height),
+//            pathEffect = pathEffect
+        )
+    }
+}
+
+
+/**
+ * Draw X and Y axes.
+ *
+ * @param color Axes color.
+ * @param keys Keys to be plotted in the X-axis.
+ */
+private fun <T> DrawScope.drawAxes(
+        color: Color,
+        keys: List<T>
+) {
+    val startPoint = origin()
+
+    // X-axis
+    drawLine(
+            color = color,
+            start = startPoint,
+            end = Offset(size.width, startPoint.y)
+    )
+    // Y-axis
+    drawLine(
+            color = color,
+            start = startPoint,
+            end = Offset(0f, 0f)
+    )
+
+    drawIntoCanvas {
+        val valuePerGrid = mMaxVal / mMaxLen
+
+        val valTextPaint = Paint()
+        valTextPaint.textAlign = Paint.Align.RIGHT
+        valTextPaint.textSize = AXIS_VALUES_FONT_SIZE
+        valTextPaint.color = 0xff000000.toInt()
+
+        val keyTextPaint = Paint()
+        keyTextPaint.textAlign = Paint.Align.CENTER
+        keyTextPaint.textSize = AXIS_VALUES_FONT_SIZE
+        keyTextPaint.color = 0xff000000.toInt()
+
+        keys.forEachIndexed { i, key ->
+            val valOffset = Offset(
+                    -20f,
+                    (startPoint.y - (i + 1) * mVGap) + AXIS_VALUES_FONT_SIZE / 2
+            )
+            val keyOffset = Offset(
+                    (i + 1) * mHGap,
+                    startPoint.y + AXIS_VALUES_FONT_SIZE + 10f
+            )
 
+            it.nativeCanvas.drawText(
+                    (valuePerGrid * (i + 1)).toInt().toString(),
+                    valOffset.x,
+                    valOffset.y,
```

Comment:
if possible make a function for this

Reply:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.